src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs

```diff
@@ -198,6 +206,60 @@ impl HummockVersionExt for HummockVersion {
             .map(|group| group.levels.len() + 1)
             .unwrap_or(0)
     }
+
+    fn calculate_compaction_group_statistic(&self) -> Vec<TableGroupInfo> {
+        let mut infos = vec![];
+        for (group_id, group) in &self.levels {
+            let group_size = group
+                .levels
+                .iter()
+                .map(|level| level.total_file_size)
+                .sum::<u64>()
+                + group.l0.as_ref().unwrap().total_file_size;
+            // only calculate the bottommost too level.
+            let mut table_statistic: HashMap<StateTableId, u64> = HashMap::new();
+            let member_table_id: HashSet<u32> = HashSet::from_iter(group.member_table_ids.clone());
+            for level in &group.l0.as_ref().unwrap().sub_levels {
+                if level.level_type() == LevelType::Overlapping {
+                    continue;
+                }
+                for sst in &level.table_infos {
+                    if sst.table_ids.len() > 1 {
+                        // do not calculate size for small state-table.
+                        continue;
+                    }
+                    if !member_table_id.contains(&sst.table_ids[0]) {
+                        continue;
+                    }
+                    let entry = table_statistic.entry(sst.table_ids[0]).or_default();
+                    *entry += sst.file_size;
+                }
+            }
+
+            for level in &group.levels {
+                for sst in &level.table_infos {
+                    if sst.table_ids.len() > 2 {
```

Comment:
Is it possible to standardize the conditions of l0 and other levels? It would be logically clearer.

Reply:
ok